packages/hardhat/test/integration-tests/crabv2/crab-timelock.ts

```diff
@@ -0,0 +1,209 @@
+    const ethToDeposit = ethers.utils.parseUnits('20')
+    const msgvalue = ethers.utils.parseUnits('10.2')
+    const depositorSqueethBalanceBefore = await wSqueeth.balanceOf(depositor.address)
+
+    await crabStrategy.connect(depositor).flashDeposit(ethToDeposit, { value: msgvalue })
+
+    const currentScaledSquethPrice = (await oracle.getTwap(wSqueethPool.address, wSqueeth.address, weth.address, 300, false))
+    const feeRate = await controller.feeRate()
+    const ethFeePerWSqueeth = currentScaledSquethPrice.mul(feeRate).div(10000)
+    const squeethDelta = scaledStartingSqueethPrice1e18.mul(2);
+    const debtToMint = wdiv(ethToDeposit, (squeethDelta.add(ethFeePerWSqueeth)));
+    const expectedEthDeposit = ethToDeposit.sub(debtToMint.mul(ethFeePerWSqueeth).div(one))
+
+    const totalSupply = (await crabStrategy.totalSupply())
+    const depositorCrab = (await crabStrategy.balanceOf(depositor.address))
+    const strategyVault = await controller.vaults(await crabStrategy.vaultId());
+    const debtAmount = strategyVault.shortAmount
+    const depositorSqueethBalance = await wSqueeth.balanceOf(depositor.address)
+    const strategyContractSqueeth = await wSqueeth.balanceOf(crabStrategy.address)
+    const lastHedgeTime = await crabStrategy.timeAtLastHedge()
+    const currentBlockNumber = await provider.getBlockNumber()
+    const currentBlock = await provider.getBlock(currentBlockNumber)
+    const timeStamp = currentBlock.timestamp
+    const collateralAmount = await strategyVault.collateralAmount
+
+    expect(isSimilar(totalSupply.toString(), (expectedEthDeposit).toString())).to.be.true
+    expect(isSimilar(depositorCrab.toString(), expectedEthDeposit.toString())).to.be.true
+    expect(isSimilar(debtAmount.toString(), debtToMint.toString())).to.be.true
+    expect(depositorSqueethBalance.eq(depositorSqueethBalanceBefore)).to.be.true
+    expect(strategyContractSqueeth.eq(BigNumber.from(0))).to.be.true
+    expect(lastHedgeTime.eq(timeStamp)).to.be.true
+  })
+
+  before("Deposit into strategy", async () => {
+    await crabStrategy.connect(depositor3).deposit({ value: ethers.utils.parseUnits('1') })
+  })
+
+  describe("Transfer vault with timelock", async () => {
+    let txHash = ''
+    let eta = 0
+    let currentBlockNumber = 0
+    let currentBlockTimestamp = 0
+    let data = ''
+
+    this.beforeAll(async () => {
+      currentBlockNumber = await provider.getBlockNumber()
+      const currentBlock = await provider.getBlock(currentBlockNumber)
+      eta = currentBlock.timestamp + twoDays + 300
+      currentBlockTimestamp = currentBlock.timestamp
+
+      data = abi.encode(['address'], [depositor2.address]);
+
+      txHash = ethers.utils.keccak256(abi.encode(['address', 'uint256', 'string', 'bytes', 'uint256'],
+        [crabStrategy.address, '0', signature, data, eta]))
+      await timelock.connect(owner).queueTransaction(crabStrategy.address, 0, signature, data, eta)
```

Comment:
could you make this a test instead of in the before all and check that it doesn't work if we accidentally pass in bad data?